File: Tomida2015/plot_func.py
import numpy as np
import matplotlib.pyplot as plt

def timecourse(sim):
    fig=plt.figure(figsize=(4,12))
    plt.rcParams['font.family'] = 'Arial'
    plt.rcParams['font.size'] = 15
    plt.rcParams['axes.linewidth'] = 2
    plt.rcParams['lines.linewidth'] = 2

    plt.subplots_adjust(wspace=0.5, hspace=0.3)

    for i in range(sim.condition):
        plt.subplot(4,1,i+1)

        plt.gca().spines['right'].set_visible(False)
        plt.gca().spines['top'].set_visible(False)
        plt.gca().yaxis.set_ticks_position('left')
        plt.gca().xaxis.set_ticks_position('bottom')

        plt.plot(sim.t,np.zeros(len(sim.t)),'--',color='darkgray')
        plt.plot(sim.t,sim.p38_activity[:,i],label='p38 reporter')

        plt.xlim([-20,480])
        plt.xticks([60*i for i in range(9)])
        plt.ylim([-0.05,0.6])
        plt.yticks([0.1*i for i in range(7)])
        if i==0:
                plt.legend(loc='upper right',frameon=False)
        fig.text(0.5, 0.08, 'Time (min)', ha='center')
        fig.text(-0.05, 0.5, 'p38 activity (a.u.)', va='center', rotation='vertical')

    fig.text(0.5, 0.82, r'$k_8$ = 0.024', ha='center')
    fig.text(0.5, 0.62, r'$k_8$ = 0.012', ha='center')
    fig.text(0.5, 0.42, r'$k_8$ = 0.008', ha='center')
    fig.text(0.5, 0.22, r'$k_8$ = 0.004', ha='center')

    plt.savefig('p38_activity.png',bbox_inches='tight')

File: Yang2018/plotFunc.jl
function plotFunc_timecourse(Sim::Module)
    rc("figure",figsize = (10.5,4));
    rc("font",family = "Arial");
    rc("font",size = 12);
    rc("axes",linewidth = 1);
    rc("lines",linewidth = 2);

    subplot(1,2,1);
    plot(Sim.t,Sim.p53_conc[:,1],"k",label="1μM etoposide");
    plot(Sim.t,Sim.p53_conc[:,2],"r",label="100μM etoposide");
    xlim(0,48);
    xticks([0,12,24,36,48]);
    ylim(0,8);
    yticks([1,3,5,7]);
    xlabel("Time (hours)");
    ylabel("[p53]");
    legend(loc="upper right",frameon=false);
    text(1,7.5,"U-2 OS");

    subplot(1,2,2);
    plot(Sim.t,Sim.p53_conc[:,3],"k",label="5μM etoposide");
    plot(Sim.t,Sim.p53_conc[:,4],"r",label="200μM etoposide");
    xlim(0,48);
    xticks([0,12,24,36,48]);
    ylim(0,5.5);
    yticks([0,1,2,3,4,5]);
    xlabel("Time (hours)");
    ylabel("[p53]");
    legend(loc="upper right",frameon=false);
    text(1,5.15,"MCF7")
    
    show();

end
